### PR TITLE
.Net: Update StepwisePlannerConfig to Improve MaxPromptTokens Calculation and Add Tests for Edge Cases

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
This pull request updates the StepwisePlannerConfig class in the Planning.StepwisePlanner project to improve the calculation of the MaxPromptTokens property, ensuring that the maximum number of iterations allowed in a plan is properly taken into account. Additionally, it introduces a new configuration property for the maximum number of tokens allowed in a completion request and adds two new tests to cover edge cases where the number of functions is too large or almost too large.

Changes:
- Modified the MaxPromptTokens property in the StepwisePlannerConfig class to use the MaxTokensRatio for calculating the maximum number of prompt tokens.
- Updated the summary comment for the MaxPromptTokens property to reflect the new calculation method.
- Add a new test `ExecutePlanFailsWithTooManyFunctions` to check if the execution fails when there are too many functions.
- Add a new test `ExecutePlanSucceedsWithAlmostTooManyFunctions` to check if the execution succeeds when there are almost too many functions.
- Update `StepwisePlannerConfig` to include a new property `MaxTokensRatio` for controlling the ratio of tokens allocated to the completion request.
- Update `StepwisePlanner` to use the new `MaxCompletionTokens` property for setting the maximum number of tokens allowed in a completion request.
- Modify the trimming logic in `StepwisePlanner` to account for the new token handling configuration and throw an exception if the chat history is too long.
- Update the `TrimMessageFormat` constant in `StepwisePlanner` to include the number of steps removed in the message.
- No other files or classes were affected by these changes.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs [hunk 1]
```diff
@@ -96,6 +98,49 @@ public async void CanExecuteStepwisePlan(bool useChatModel, string prompt, strin
...
+        Assert.Equal("ChatHistory is too long to get a completion. Try reducing the available functions.", ex.Message);
+    }
+
+    [Fact]
+    public async void ExecutePlanSucceedsWithAlmostTooManyFunctions()
+    {
+        // Arrange
+        IKernel kernel = this.InitializeKernel();
+
+        _ = await kernel.ImportAIPluginAsync("Klarna", new Uri("https://www.klarna.com/.well-known/ai-plugin.json"), new OpenApiSkillExecutionParameters(enableDynamicOperationPayload: true)).ConfigureAwait(false);
```
Comment (lemillermicrosoft) [inline dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:131]: Will change this to something with a bit more control -- this was just the available repro.

File: dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlannerConfig.cs [hunk 2]
```diff
@@ -51,12 +51,21 @@ public sealed class StepwisePlannerConfig
     #region Execution configuration
 
     /// <summary>
-    /// The maximum number of tokens to allow in a request and for completion.
+    /// The maximum number of total tokens to allow in a completion request (prompt + completion)
```
Comment (hario90) [inline dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlannerConfig.cs:54]: nit:
```suggestion
    /// The maximum total number of tokens to allow in a completion request, which includes the tokens from the prompt and completion
```

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs [hunk 3]
```diff
@@ -96,6 +98,49 @@ public async void CanExecuteStepwisePlan(bool useChatModel, string prompt, strin
...
+
+        var planner = new Microsoft.SemanticKernel.Planning.StepwisePlanner(kernel);
+
+        // Act
+        var plan = planner.CreatePlan("I need to buy a new brush for my cat. Can you show me options?");
+        var result = await kernel.RunAsync(plan);
+
+        // Assert - should contain results, for now just verify it didn't fail
+        Assert.NotNull(result.Result);
+        Assert.DoesNotContain("Result not found, review 'stepsTaken' to see what happened", result.Result, StringComparison.OrdinalIgnoreCase);
```
Comment (hario90) [inline dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:141]: Consider making the error message a string accessible here so that if the error message changes, it doesn't need to be updated here as well.

**Review by hario90 — APPROVED** [review]

File: dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs (hunk 3 above)
Comment (lemillermicrosoft) [reply dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:141]: Agreed. However the shared dependencies on things make this hard to do right now with exposing internals, sadly. Hoping the folder restructuring will make this easier.

Comment (hario90) [reply dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs:142]: Ah, makes sense.

**Review by shawncal — APPROVED** [review]
